numpy_ml/linear_models/lm.py

```diff
@@ -27,6 +27,41 @@ def __init__(self, fit_intercept=True):
         """
         self.beta = None
         self.fit_intercept = fit_intercept
+        self.inverse_cov = None
+
+    def update(self, x, y):
+        """
+        Incrementally update the regression coefficients using Matrix Inversion Lemma RLS, version 1 ( https://github.com/ddbourgin/numpy-ml/files/6536069/rls.pdf ).
+
+        Parameters
+        ----------
+        x : :py:class:`ndarray <numpy.ndarray>` of shape `(N, M)` on first attempt and shape (1, M) on subsequent updates
+            A dataset consisting of `N` examples, each of dimension `M`.
+        y : :py:class:`ndarray <numpy.ndarray>` of shape `(N, K)` on first attempt and shape (1, K) on subsequent updates
+            The targets for each of the `N` examples in `X`, where each target
+            has dimension `K`.
+        """
+        xm = np.mat(x)
+        ym = np.mat(y)
+
+        if self.fit_intercept:
+            xm = np.c_[np.ones(xm.shape[0]), xm]
+
+        all_zeros_or_empty_inverse_cov = not np.any(self.inverse_cov)
+        all_zeros_or_empty_beta = not np.any(self.beta)
+
+        if all_zeros_or_empty_inverse_cov or all_zeros_or_empty_beta: # first run of the algorithm
```

Comment:
I'm wondering if we should allow users to fit the initial model via the `update` method. That feels like it doesn't align with the semantics of the term "update." I can think of two possible resolutions: 

1. Throw some kind of runtime exception if we call `update` on a model that hasn't already been fit

```python
if not self.is_fit:
    raise RuntimeError("You must call the `fit` method before calling `update`)
```

2. Introduce a `fit_incremental` method and make `update` a private method. Here, `fit_incremental` would internally call `fit` if the model params aren't initialized, or `_update` if they are.

To me, (1) feels preferable since otherwise we're left with two different public fit methods, which feels a little confusing. Thoughts?